warrior/Framework/Utils/data_Utils.py

```diff
@@ -1394,7 +1394,7 @@ def subst_var_patterns_by_prefix(raw_value, start_pattern="${",
             if len(extracted_var) > 0:
                 for string in extracted_var:
                     try:
-                        if isinstance(raw_value[k], str):
+                        if isinstance(raw_value[k], (str, unicode)):
```

Comment:
could be checked for the instance of basestring which would handle both str and unicode.